### DISCUSSION TITLE
Question about generated download links.

### DISCUSSION THREAD

**Discussion by r0hin** [fent/node-ytdl-core#1000] [Q&A] [answered]
I've noticed that the following restriction: `Generated download links are valid for 6 hours, and may only be downloadable from the same IP address.` applies to a subset of videos. My question is: Are there any other **IP address-related** restrictions on videos that fall under this category. 

For example, if I notice that a generated URL works on two separate IP addresses, do you know if it will continue to work on more IP addresses or is there potentially a hard cap on maybe 10 IP addressed for example.

Thanks for your help! (And could someone take a look at #985?)

**Comment by TimeForANinja** [comment, marked as answer]
> do you know if it will continue to work on more IP addresses

it should

the url's contain an ip parameter and a signature so that you don't just change the ip
if that url is `a` and you successfully request from `b` it will work for `c`

**Comment by gatecrasher777** [comment]
I have noticed that no IP restrictions apply if the n-parameter challenge is solved.  The transformed urls seem to work from any IP address.

**Reply by r0hin** [reply]
That seems very interesting. Could you please provide an example of what you mean?

**Comment by gatecrasher777** [comment]
Using an app I'm developing that depends on [ytcog](https://github.com/gatecrasher777/ytcog) I use a VPN (located in a different country) on one machine which does the bulk of the work fetching video stream data. But some videos require a login, so I have another version of the app on another machine (via my ISP) which is logged into my YouTube account. Any video info request that fails in the first app is retrieved via the logged-in app and sent back to the first app.  These retrieved streams download fine just via the VPN (in a different country to where the links are retrieved). And it doesn't seem to matter where the VPN is located. 

Two differences to the current ytdl-core streams is that 1) the streams are requested via the innertube API vs the Watch Page and 2) the n-parameter challenge is solved.  I'm inclined to think that the solved n-challenge validates the stream request on the Google servers and causes the IP check to be skipped. But I could be wrong. All I know for sure is that it works. 

I don't know if those streams will work from 10 or more locations though. You would have to experiment.

**Comment by lockieluke** [comment]
so briefly how do i bypass the ip check

**Reply by gatecrasher777** [reply]
In my experience, since YouTube introduced the n parameter, there is no IP check. Once you have a generated link (deciphered if necessary with the n transform), you can use it to download from any IP address. 

I'm not sure, however, if this behaviour is different under different circumstances. In my case I use a logged-in cookie and info from [ytcog](https://github.com/gatecrasher777/ytcog) innertube requests.
